metagraph/src/cli/load/load_annotated_graph.cpp

```diff
@@ -19,7 +18,7 @@ using mtg::common::logger;
 std::unique_ptr<AnnotatedDBG> initialize_annotated_dbg(std::shared_ptr<DeBruijnGraph> graph,
                                                        const Config &config) {
     // TODO: check and wrap into canonical only if the graph is primary
-    if (config.canonical && !graph->is_canonical_mode())
+    if (config.canonical && !graph->is_canonical_mode() && config.identity != Config::ASSEMBLE)
```

Comment:
Too much digging for this loader.

Better pass a flag as a parameter:
`initialize_annotated_dbg(... graph, ... config, bool wrap_primary = true);`

Reply:
doing it this way is kind of problematic because if `wrap_primary == false` and one passes a primary graph, we still need to wrap it and this function then ends up returning something inexpected